homeassistant/components/zwave_js/humidifier.py

```diff
@@ -0,0 +1,214 @@
+
+        if (
+            str(HumidityControlMode.HUMIDIFY.value)
+            in info.primary_value.metadata.states
+        ):
+            entities.append(ZWaveHumidifier(config_entry, client, info))
+
+        if (
+            str(HumidityControlMode.DEHUMIDIFY.value)
+            in info.primary_value.metadata.states
+        ):
+            entities.append(ZWaveDehumidifier(config_entry, client, info))
+
+        async_add_entities(entities)
+
+    config_entry.async_on_unload(
+        async_dispatcher_connect(
+            hass,
+            f"{DOMAIN}_{config_entry.entry_id}_add_{HUMIDIFIER_DOMAIN}",
+            async_add_humidifier,
+        )
+    )
+
+
+class ZWaveBaseHumidifier(ZWaveBaseEntity, HumidifierEntity):
+    """Representation of a Z-Wave Humidifier or Dehumidifier."""
+
+    _current_mode: ZwaveValue
+    _setpoint: ZwaveValue | None = None
+
+    _on_mode = HumidityControlMode
+    _inverse_mode = HumidityControlMode
+
+    def __init__(
+        self,
+        config_entry: ConfigEntry,
+        client: ZwaveClient,
+        info: ZwaveDiscoveryInfo,
+    ) -> None:
+        """Initialize thermostat."""
+        super().__init__(config_entry, client, info)
+
+        self._attr_name = f"{self._attr_name} {self.device_class}"
+        self._attr_unique_id = f"{self._attr_unique_id}_{self.device_class}"
+
+        self._current_mode = self.get_zwave_value(
+            HUMIDITY_CONTROL_MODE_PROPERTY,
+            command_class=CommandClass.HUMIDITY_CONTROL_MODE,
+        )
+
+        if not self._current_mode:
+            raise ValueError("Humidity control mode is required")
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return True if entity is on."""
+        return int(self._current_mode.value) in [
+            self._on_mode,
+            HumidityControlMode.AUTO,
+        ]
+
+    def _supports_inverse_mode(self) -> bool:
+        return str(self._inverse_mode.value) in self._current_mode.metadata.states
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on device."""
+        mode = int(self._current_mode.value)
+        if mode == HumidityControlMode.OFF:
+            new_mode = self._on_mode
+        elif mode == self._inverse_mode:
+            new_mode = HumidityControlMode.AUTO
+        else:
+            return
+
+        await self.info.node.async_set_value(self._current_mode, new_mode)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off device."""
+        mode = int(self._current_mode.value)
+        if mode == HumidityControlMode.AUTO:
+            if self._supports_inverse_mode():
+                new_mode = self._inverse_mode
+            else:
+                new_mode = HumidityControlMode.OFF
+        elif mode == self._on_mode:
+            new_mode = HumidityControlMode.OFF
+        else:
+            return
+
+        await self.info.node.async_set_value(self._current_mode, new_mode)
```

Comment:
turn_on and turn_off implement this logic: https://github.com/home-assistant/core/pull/65847#discussion_r800217658